clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -255,11 +257,15 @@ public long maximumTimeToWait(long currentTimeMs) {
      * member to {@link MemberState#JOINING}, so that it rejoins the group.
      */
     public void resetPollTimer(final long pollMs) {
+        pollTimer.update(pollMs);
         if (pollTimer.isExpired()) {
```

Comment:
I would rather have a method added to `Timer` such as `long hasExpiredBy()` so the check for expiration and the calculation of by how much is encapsulated in the timer itself.

Reply:
agree, makes total sense, so moved the calculation to the timer, with an `isExpiredBy`. Small twist to what I understand you were suggesting, I kept the `isExpired` check, just to avoid having to deal with the logic of deducing if the timer is expired based on the `isExpiredBy` on the HBManager. Seems better to let the timer know the semantics of when it's considered expired (it does consider >= for instance, so just avoiding to bring those semantics into the HBManager). Makes sense?

Reply:
Yes, makes sense. When I was reviewing the previous iteration, I found myself looking within the Timer at the internal variables and then trying to figure out whether the derivation being performed was valid. Makes sense to do it within the Timer. Perfectly happy with 2 methods like this.